### PR TITLE
feat(licenseinfo): allow users to edit the license info header text

### PR DESCRIPTION
**Pull request by ChristophNiehoff**
- At the project edit: The 'Administration' tab now allows to edit a custom header for the license info output.
- This header is included to the license info outout for txt-, xhtml- and docx-files.
- To clean up the project edit page, tabbing (as for the release edit page) was introduced
- `projects.thrift` now contains an additional field `licenseInfoHeaderText`. If this is unset then the default text (located at `backend/src/src-licenseinfo/src/main/resources/DefaultLicenseInfoHeader.txt`) will be used. If the license info header text was customized, then it is stored in the database at the project.

Closes sw360/sw360portal#558

### CODE REVIEW

**Comment by ChristophNiehoff** [comment]
@adrian-evo  As this PR contains many modifications to the project edit page, could you test it?

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 1]
```diff
@@ -888,6 +892,7 @@ private void prepareProjectEdit(RenderRequest request) {
 
             request.setAttribute(PROJECT, project);
             request.setAttribute(DOCUMENT_ID, id);
+            request.setAttribute(DEFAULT_LICENSE_INFO_HEADER_TEXT, StringEscapeUtils.escapeJava(getProjectDefaultLicenseInfoHeaderText()));
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: As discussed, we do need to display the default text to the user when none is set. So, never mind my comment about that.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -1063,4 +1070,29 @@ private void updateVulnerabilityRating(ResourceRequest request, ResourceResponse
...
+        if(!project.isSetLicenseInfoHeaderText()) {
+            String defaultLicenseInfoHeaderText = getProjectDefaultLicenseInfoHeaderText();
+            project.setLicenseInfoHeaderText(defaultLicenseInfoHeaderText);
+        }
+    }
+
+    private String getProjectDefaultLicenseInfoHeaderText() {
+        String defaultLicenseInfoHeader = "";
+        try {
+            defaultLicenseInfoHeader = (new LicenseInfoHandler()).getDefaultLicenseInfoHeaderText();
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: As discussed, please make the resource available via a thrift call instead

**Comment by ChristophNiehoff** [comment]
@alexbrdn  Introduced a thrift-call to get the default text from the backend to the UI. Further,  comparing the input text to the default text is now done in the javascript code.  Please have a re-review.

And also a further remark (also to @mcjaeger ): At the current state of the code the "Generate License Info" button as well as the "Linked releases" are not working (see https://github.com/sw360/sw360portal/issues/592). As already discussed, this seems to be due to problems  with `requirejs`. For the last few hours I tried to fix it, but this is like opening pandora's box. Errors similar to the one in https://github.com/sw360/sw360portal/issues/594 appear and also the mixing of javascript and JSP does not make things easier. It turns out that it is rather involved to reach a state where all javascript code is working and that many different files have to be modified. I stopped at a point where I realized that I also would have to touch the detail pages of components and releases.
Since this ticket is (kind of) independent of the above issues, I would not include them into this PR. Maybe we should consider opening an independent issue for consistently converting the javascript code to requirejs (but this is rather involved and implies a large amount of testing). We should discuss this in the next telko...

**Comment by adrian-evo** [comment]
Moderation is not working. The proposed change for this field is empty on moderation request.

Additionally, cannot generate license due to #592 

lic header

**Comment by mcjaeger** [comment]
@ChristophNiehoff maybe because we have merged #599, a rebase may help to overcome also the requirejs problems.

**Comment by ChristophNiehoff** [comment]
updated to the current master and fixed some minor issues.
@adrian-evo  could you please retest?

**Comment by adrian-evo** [comment]
In both scenarios below, approving the moderation is working, however the displayed text on moderation tab is not always right.

Scenario1:
- Login as user@sw360, create a project, modify default header to e.g. "New header text..." 
- Login as user1@sw360, create a moderation request by pressing the button Set To Default Text.
- Login as user@sw360, check moderation. Expected is that Suggested Value display the default text. Actual is that the suggested values is "--not set--".

Scenario2:
- Login as user@sw360, create a project
- Login as user1@sw360, create a moderation request by changing the field to e.g. "New header text..."
- Open the moderation from My Task Submissions. License Info Header contains no text, expected is that previous text is there.

The following seems to be reproducible in a clean cache browser in Firefox, and also I noticed it from time to time, not clear in which context.

- Login as user@sw360, create a project
- Login as user1@sw360, edit to create a moderation request
- See attached, License Info Header is empty and also Set To Default Text Button is not working

header

**Comment by mcjaeger** [comment]
@adrian-evo @ChristophNiehoff 
I think **scenario 1** is OK, because, if there would be a text (which is the default) the user may have a hard time to check if this is the default text and not changed (maybe even a change can be in very few words) so a resultion for this could be to write "-- default text --" instead of "not set"

**Scenario 2** is maybe necessary to see the text for the moderator in order to be able to decide on approval. So, it would be better to actually display the text there.

**Comment by ChristophNiehoff** [comment]
@mcjaeger 
I had another idea:  In both cases (scenario 1 and 2): "--default text--" is shown but the default text itself is visible as a hover text. With this it is always clear whether the default text is set, but for making a decision the moderator can see what it actually is.
Example for changing some custom text back to the default one:
defaulttext
What do you think?

@adrian-evo Sorry, I still cannot reproduce your 3rd remark.   Are there any javascript-errors thrown in the browser console?  @mcjaeger  Can you reproduce this?

**Comment by mcjaeger** [comment]
testing is Ok from my side. @alexbrdn also Ok with your requested changes?

File: backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java [hunk 3]
```diff
@@ -257,4 +267,15 @@ protected LicenseInfoParsingResult noSourceParsingResult() {
         return outputGenerators.stream().filter(outputGenerator -> generatorClassname.equals(outputGenerator.getClass().getName()))
                 .findFirst().orElseThrow(() -> new TException("Unknown output generator: " + generatorClassname));
     }
+
+    private static String loadDefaultLicenseInfoHeaderText(){
+        try {
+            String defaultLicenseInfoHeader = IOUtils.toString(LicenseInfoHandler.class.getResourceAsStream(DEFAULT_LICENSE_INFO_HEADER_FILE), "UTF-8");
```
Comment (alexbrdn) [inline backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java]: You have not changed to use `CommonUtils.loadResource()` as requested. I still think it is rather important to do so.

**Comment by ChristophNiehoff** [comment]
@alexbrdn Implemented the requested change. Please have a re-review.

**Pull request merged**